src/zenml/zen_stores/schemas/tag_schemas.py

```diff
@@ -123,14 +126,21 @@ class TagResourceSchema(BaseSchema, table=True):
         back_populates="tags",
         sa_relationship_kwargs=dict(
             primaryjoin=f"and_(TagResourceSchema.resource_type=='{TaggableResourceTypes.ARTIFACT.value}', foreign(TagResourceSchema.resource_id)==ArtifactSchema.id)",
-            overlaps="tags,model",
+            overlaps="tags,model,artifact_version",
+        ),
+    )
+    artifact_version: List["ArtifactVersionSchema"] = Relationship(
+        back_populates="tags",
+        sa_relationship_kwargs=dict(
+            primaryjoin=f"and_(TagResourceSchema.resource_type=='{TaggableResourceTypes.ARTIFACT_VERSION.value}', foreign(TagResourceSchema.resource_id)==ArtifactVersionSchema.id)",
+            overlaps="tags,model,artifact",
```

Comment:
What does this do?

Reply:
tags are linked with other entities via a single table `tag_resource` which has columns `tag_id` and `resource_id`, where`resource_id` can refer to a variety of different entities (i.e. it's not a standard foreign key constraint in the DB and this is how you specify such a shared constraint column in SQLModel. See #2036 for more info.